### PR TITLE
Schema Adjustments + Cleanups

### LINKED ISSUE #24: `maxLength` and `minLength` properties have wrong data type
__Describe the Bug__

[Inside the schema file](https://github.com/camunda/element-templates-json-schema/blob/master/resources/schema.json#L457-L468), the `maxLength` and `minLength` properties of a constraint have the data type `String`, which is wrong.

Following (originally correct) constraint would be wrong currently

```json
"constraints": {
  "mandatory": true,
  "minLength": 10,
  "maxLength": 20,
  "pattern": {
    "message": "Must be valid shoe size (>20 && <49)",
    "match": "/[2-4][0-9]/"
  }
}
```

__Steps to Reproduce__

1. Create a template with a property constraint as shown above
2. Validate template against the schema
3. Schema is not valid

```js
{
  keyword: 'type', 
  dataPath: '/properties/0/constraints/minLength', 
  schemaPath: '#/definitions/properties/items/properties/constraints/properties/minLength/type', 
  params: Object{type: ...}, 
  message: 'should be string'
}, 
{
  keyword: 'type', 
  dataPath: '/properties/0/constraints/maxLength', 
  schemaPath: '#/definitions/properties/items/properties/constraints/properties/maxLength/type', 
  params: Object{type: ...}, 
  message: 'should be string'
}
```

__Expected Behavior__

__Environment__

- Host (Browser/Node version), if applicable: Node v14.16.0
- OS: macOS v10.15
- Library version: 0.3.0 (and all below)

### LINKED ISSUE #25: Incomplete error message for invalid property binding type
__Describe the Bug__

When validating the following example template

```json
[
  {
    "name": "Invalid",
    "id": "foo",
    "appliesTo": [
      "bpmn:UserTask"
    ],
    "properties": [
      {
        "label": "Label",
        "type": "String",
        "value": "YEY YEA!",
        "binding": {
          "type": "alsoInvalid",
          "name": "label"
        }
      }
    ]
  }
]
```

the following error message appears

```error
"invalid property.binding type "alsoInvalid"; must be any of { property, camunda:property, camunda:inputParameter, camunda:outputParameter, camunda:in, camunda:out, camunda:in:businessKey, camunda:executionListener, camunda:field }"
```

Note that the `camunda:errorEventDefinition` binding is missing in the error message. That's is basically missing in the [defined `errorMessage` for this rule](https://github.com/camunda/element-templates-json-schema/blob/master/resources/schema.json#L386).

__Steps to Reproduce__

1. Take the example template from above
2. Validate against JSON Schema
3. Note incorrect error message

__Expected Behavior__

Error messages labels

```error
"invalid property.binding type "alsoInvalid"; must be any of { property, camunda:property, camunda:inputParameter, camunda:outputParameter, camunda:in, camunda:out, camunda:in:businessKey, camunda:executionListener, camunda:field, camunda:errorEventDefinition }"

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Fix smaller Schema issues while working on camunda/camunda-modeler#2159, especially to align with the existing Properties Panel behavior. That includes
* correct data types for `constraints.{minLength | maxLength}` properties (cf. #24)
* adjust error message for invalid binding type (cf. #25)
* adjust error messages for required properties
* adjust error message for legacy scope type
* adjust error message for `appliesTo` pattern
* fix a test case for template versions
* add additional allowed data types for `entriesVisible` & `property.constraints.pattern`
* add more complex test cases

----

Related to camunda/camunda-modeler#2159
Closes #24 
Closes #25

### CODE REVIEW

File: resources/schema.json [hunk 1]
```diff
@@ -614,7 +633,10 @@
           "items": {
             "$id": "#/appliesTo/items",
             "type": "string",
-            "pattern": "^(bpmn:)"
+            "pattern": "^(bpmn:)",
+            "errorMessage": {
+              "pattern": "invalid item for \"appliesTo\", should match pattern ^(bpmn:)"
```
Comment (nikku) [inline resources/schema.json:638]: Can technically also match things like `camunda:ServiceTaskLike`. It should be a namespaced property name, at least.

File: test/spec/validationSpec.js [hunk 2]
```diff
@@ -132,6 +141,20 @@ describe('validation', function() {
...
+
+
+    testTemplate('invalid-applies-to');
+
+
+    testTemplate('entries-visible-boolean');
+
+
+    testTemplate('pattern-string');
+
```
Comment (nikku) [inline test/spec/validationSpec.js:157]: Nitty gritty: Add new line.

**Review by nikku — CHANGES_REQUESTED** [review]
Looks good. Great test coverage, too.

Found one quirk worth having a look at.

**Review by nikku — APPROVED** [review]

**Comment by nikku** [comment]
Good work. I'm looking forward for strictvalidation to land in the Camunda Modeler.

**Comment by nikku** [comment]
Pending release and integration into properties panel I believe.

**Comment by pinussilvestrus** [comment]
👍  I will execute the missing steps alongside https://github.com/bpmn-io/bpmn-js-properties-panel/pull/455

**Comment by nikku** [comment]
That's great :clap:

**Pull request merged**